src/plots/waterfallPlot.ts

```diff
@@ -0,0 +1,219 @@
+          var isTotal = totalAccessor(d, i, dataset);
+          if (isTotal) {
+            return Math.min(yScale.scale(currentValue), yScale.scale(0));
+          } else {
+            var currentSubtotal = this._subtotals[i];
+            if (i === 0) {
+              if (currentValue < 0) {
+                return yScale.scale(currentSubtotal - currentValue);
+              } else {
+                return yScale.scale(currentSubtotal);
+              }
+            }
+            var priorSubtotal = this._subtotals[i - 1];
+            if (currentSubtotal > priorSubtotal) {
+              return yScale.scale(currentSubtotal);
+            } else {
+              return yScale.scale(priorSubtotal);
+            }
+          }
+        };
+      }
+
+      var heightAttr = this.attr("height");
+      if (heightAttr == null) {
+        attrToProjector["height"] = (d, i, dataset) => {
+          var isTotal = totalAccessor(d, i, dataset);
+          var currentValue = this.y().accessor(d, i, dataset);
+          if (isTotal) {
+            return Math.abs(yScale.scale(currentValue) - yScale.scale(0));
+          } else {
+            var currentSubtotal = this._subtotals[i];
+            if (i === 0) {
+              return Math.abs(yScale.scale(Math.abs(currentValue)) - yScale.scale(0));
+            } else {
+              var priorSubtotal = this._subtotals[i - 1];
+              return Math.abs(yScale.scale(currentSubtotal) - yScale.scale(priorSubtotal));
+            }
+          }
+        };
+      }
+
+      attrToProjector["class"] = (d, i, dataset) => {
+        var baseClass = "";
+        if (this.attr("class") != null) {
+          baseClass = this.attr("class").accessor(d, i, dataset) + " ";
+        }
+        var isTotal = totalAccessor(d, i, dataset);
+        if (isTotal) {
+          return baseClass + Waterfall._BAR_TOTAL_CLASS;
+        } else {
+          var delta  = this.y().accessor(d, i, dataset);
+          return baseClass + (delta > 0 ? Waterfall._BAR_GROWTH_CLASS : Waterfall._BAR_DECLINE_CLASS);
+        }
+      };
+
+      return attrToProjector;
+    }
+
+    protected _onDatasetUpdate() {
+      this._updateSubtotals();
+      super._onDatasetUpdate();
+      return this;
+    }
+
+    private _calculateSubtotalsAndExtent(dataset: Dataset) {
+      var min = Number.MAX_VALUE;
+      var max = Number.MIN_VALUE;
+      var total = 0;
+      var startValue = -Infinity;
+      dataset.data().forEach((datum, index) => {
+        var currentValue = this.y().accessor(datum, index, dataset);
+        var isTotal = this.total().accessor(datum, index, dataset);
+        if (!isTotal || index === 0) {
+          total += currentValue;
+        }
+        this._subtotals.push(total);
+        if (total < min) {
+          min = total;
+        }
+        if (total > max) {
+          max = total;
+        }
+        if (startValue === -Infinity && isTotal) {
+          var adjustment = currentValue - total;
```

Comment:
`adjustment` is not very descriptive. I believe this is the offset of all deltas. Call it `startValue` or something more descriptive